polaris/benchmark/predictions/_base.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+from polaris.utils.types import IncomingPredictionsType, ListOrArrayType, PredictionsType
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    model_validator,
+    field_serializer,
+)
+
+
+class BenchmarkPredictions(BaseModel):
+    predictions: PredictionsType
+    target_cols: list[str]
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @field_serializer("predictions")
+    def _serialize_predictions(self, predictions: PredictionsType):
+        """Converts all numpy values in the predictions dictionary to lists so
+        they can be sent over the wire."""
+
+        def convert_to_list(v):
+            if isinstance(v, np.ndarray):
+                return v.tolist()
+            elif isinstance(v, dict):
+                return {k: convert_to_list(v) for k, v in v.items()}
+
+        return convert_to_list(predictions)
+
+    @model_validator(mode="before")
+    @classmethod
+    def validate_predictions(cls, data):
+        vals = data.get("predictions")
+        target_cols = data.get("target_cols")
+
+        if vals is None:
+            raise ValueError("predictions must be provided")
+
+        if target_cols is None:
+            raise ValueError("target_cols must be provided")
+
+        predictions_in_correct_shape = cls._normalize_predictions(vals, target_cols)
+        predictions_with_correct_types = cls._convert_lists_to_numpy_arrays(predictions_in_correct_shape)
+        data["predictions"] = predictions_with_correct_types
+
+        return data
+
+    @classmethod
+    def _normalize_predictions(
+        cls, vals: IncomingPredictionsType, target_cols: list[str]
+    ) -> dict[str, dict[str, ListOrArrayType]]:
+        """Normalizes the predictions so they are always in the standard format,
+        {"test-set-name": {"target-name": np.ndarray}} regardless of how they are provided."""
+
+        if isinstance(vals, (list, np.ndarray)):
+            return {"test": cls._set_col_name(vals, target_cols)}
+        elif cls._is_multi_task_single_test_set(vals, target_cols):
+            return {"test": vals}
+        elif cls._is_single_task_multi_test_set(vals, target_cols):
+            return {test_set_name: cls._set_col_name(v, target_cols) for test_set_name, v in vals.items()}
+        else:
+            cls._give_feedback(vals)
```

Comment:
**Suggestion:** Should this always be called? I believe it may now for example happen that we create a predictions object based on a non-numeric list.

Reply:
Non-numeric lists are now supported, thanks for clarifying!